### PR TITLE
refactor: make error dumber

### PR DESCRIPTION
**Pull request by UnbearableBear**
I can't wrap my head around `return Promise.reject` to handle errors plus I think it's buggy, errors (including 404) are not correctly shown somtimes. The code to make all this work seems very hard to maintain.

Anyway
https://nextjs.org/docs/advanced-features/custom-error-page#reusing-the-built-in-error-page
It's a recommended pattern to delegate the display of error to the page so I guess we could do both ?

We could keep the `err` param in `_app` and the custome `getInitialProps` of the error page AND use errorCode to handle common errors in the regular way ?

### CODE REVIEW

**Comment by lionelB** [comment]
I re-implement the with-sentry exemple and refactor the refreshToken to auth function to avoid token flush bug

File: targets/frontend/src/lib/auth/setRefreshTokenCookie.js [hunk 1]
```diff
@@ -10,7 +10,8 @@ export function setRefreshTokenCookie(res, refresh_token) {
       maxAge: (process.env.REFRESH_TOKEN_EXPIRES || 43200) * 60,
       path: "/",
       sameSite: "Strict",
-      secure: process.env.NODE_ENV === "production",
+      // secure: process.env.NODE_ENV === "production",
```
Comment (UnbearableBear) [inline targets/frontend/src/lib/auth/setRefreshTokenCookie.js:13]: ?

File: targets/frontend/src/pages/_app.js [hunk 2]
```diff
@@ -2,45 +2,49 @@ import "@reach/menu-button/styles.css";
...
-  enabled: process.env.NODE_ENV === "production",
-});
+if (process.env.NEXT_PUBLIC_SENTRY_DSN) {
+  const config = getConfig();
+  const distDir = `${config.serverRuntimeConfig.rootDir}/.next`;
+  Sentry.init({
+    dsn: process.env.NEXT_PUBLIC_SENTRY_DSN,
+    enabled: process.env.NODE_ENV === "production",
+    integrations: [
+      new RewriteFrames({
```
Comment (UnbearableBear) [inline targets/frontend/src/pages/_app.js:21]: Ca fait quoi RewriteFrames ?

Comment (lionelB) [reply targets/frontend/src/pages/_app.js:21]: https://docs.sentry.io/platforms/node/integrations/pluggable-integrations/#rewriteframes

File: targets/frontend/src/pages/_error.js [hunk 3]
```diff
@@ -1,74 +1,75 @@
...
   // indicate a bug introduced in Next.js, so record it in Sentry
-  Sentry.captureMessage(
-    `_error.js statusCode=${
-      res && res.statusCode
-    } getInitialProps missing data at path: ${asPath}`
+  Sentry.captureException(
+    new Error(`_error.js getInitialProps missing data at path: ${asPath}`)
   );
-
+  await Sentry.flush(2000);
```
Comment (UnbearableBear) [inline targets/frontend/src/pages/_error.js:65]: Ca fait quoi le .flush ?

File: targets/frontend/src/pages/user/edit/[id].js [hunk 4]
```diff
@@ -68,7 +68,7 @@ export function EditUserPage({ user }) {
       });
   }
   return (
-    <Layout title="Modifier mes informations">
+    <Layout errorCode={errorCode} title="Modifier mes informations">
```
Comment (UnbearableBear) [inline targets/frontend/src/pages/user/edit/[id].js:71]: Du coup errorCode n'est pas utilisé vu qu'on fait encore un `return Promise.reject` ?

File: targets/frontend/src/pages/index.js [hunk 5]
```diff
@@ -8,6 +9,14 @@ import { jsx, Text } from "theme-ui";
 export function IndexPage() {
   return (
     <Layout title="Home">
+      <Head>
+        {/*
+          Included so the browser doesn't hit a 404 attempting to GET
```
Comment (UnbearableBear) [inline targets/frontend/src/pages/index.js:14]: Not sure I understand this one !

File: targets/frontend/src/pages/_app.js [hunk 6]
```diff
@@ -2,45 +2,49 @@ import "@reach/menu-button/styles.css";
...
+        iteratee: (frame) => {
+          frame.filename = frame.filename.replace(distDir, "app:///_next");
```
Comment (UnbearableBear) [inline targets/frontend/src/pages/_app.js:23]: Du coup le but ici c'est de ne pas afficher l'aborscence du filesystem ?

File: targets/frontend/src/pages/_error.js (hunk 3 above)
Comment (lionelB) [reply targets/frontend/src/pages/_error.js:65]: https://docs.sentry.io/platforms/javascript/configuration/draining/

File: targets/frontend/src/pages/_app.js (hunk 6 above)
Comment (lionelB) [reply targets/frontend/src/pages/_app.js:23]: ui

File: targets/frontend/src/pages/user/edit/[id].js (hunk 4 above)
Comment (lionelB) [reply targets/frontend/src/pages/user/edit/[id].js:71]: ah oui j'ai du oublié de l'enlever

**Pull request merged**